src/EncryptedFS.ts

```diff
@@ -89,6 +89,34 @@ class EncryptedFS {
         },
         logger: logger.getChild(DB.name),
       });
+      // Perform db key validation
+      const keyValidationDB = await db.level('keyValidation');
+      try {
+        const canary = await keyValidationDB.get('canary-key');
+        // Try to decrypt the stored value
+        if (!(await utils.decrypt(dbKey!, utils.toArrayBuffer(canary)))) {
+          throw new errors.ErrorEncryptedFSKey();
+        }
+      } catch (e) {
```

Comment:
Separate out this logic:

```ts
// uses the root level that's already available
try {
  const deadbeefData: Buffer = await db.db.get('canary');

  try {
    const deadbeef = await db.deserializeDecrypt(deadbeefData, false);
    deadbeef === 'deadbeef';
  } catch(e) {
    // see what happens here, try different things
  }

} catch (e) {
  if (e.notFound) {
    // it doesn't exist, so you just put it and proceed
    await db.put([], 'canary', 'deadbeef');
  }
  throw e;
}
```

This initial canary check can be a separate protected function. You can call it `setupCanary`.

Reply:
I made it a static function because this is being done inside the static create function

Reply:
fixed now